packages/amplify_datastore/ios/Classes/types/model/FlutterAuthRule.swift

```diff
@@ -0,0 +1,96 @@
+/*
+* Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+
+import Foundation
+import Amplify
+
+public struct FlutterAuthRule {
+    
+    private let ownerField : String?
+    private let identityClaim : String?
+    private let groupClaim : String?
+    private let groups : [String]?
+    private let groupsField : String?
+    private var authStrategy : AuthStrategy!
+    private var operations : [ModelOperation]?
+    
+    init(serializedData: [String: Any]) {
+        
+        
+        self.ownerField = serializedData["ownerField"] as? String
+
+        self.identityClaim = serializedData["identityClaim"] as? String
+        
+        self.groupClaim = serializedData["groupClaim"] as? String
+        
+        self.groups = serializedData["groups"] as? [String]
+        
+        self.groupsField = serializedData["groupsField"] as? String
+        
+        self.authStrategy = stringToAuthStrategy(string: serializedData["authStrategy"] as! String)
+
+        self.operations =
+            (serializedData["operations"] as! [String]).map {
+                stringToModelOperation(string: $0)
+        }
```

Comment:
avoid forcecasts in favor of guards

Reply:
code updated

Reply:
```swift
guard let authStrategy = stringToAuthStrategy(string: serializedData["authStrategy"] as! String) else {
        preconditionFailure("Could not create a AuthStrategy from serializedData")
}
self.authStrategy = authStrategy
```

cc: @royjit if there is a better way for this.

Reply:
hmm were you able to get this code to compile?  I used a slightly modified version but am getting compile errors : Initializer for conditional binding must have Optional type, not 'String'

What is the benefit of this approach? 

`        guard let authStrategy = serializedData["authStrategy"] as! String else { preconditionFailure("Could not create a AuthStrategy from serializedData")
        }
        self.authStrategy = authStrategy`

Reply:
I think you need to change `as!` to `as?` to conditional unwrap/cast if possible else throw exception.
The benefit is graceful exception/error handling than runtime cast failed.